server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java

```diff
@@ -273,7 +273,7 @@ public final boolean isDone() {
      * @throws IllegalStateException if this listener is not complete yet and assertions are disabled.
      */
     @SuppressWarnings({ "unchecked", "rawtypes" })
-    protected final T rawResult() throws Exception {
```

Comment:
Bit of a quick and dirty solution for now to be able to detect a done and failed listener. I didn't want to add an API for this just yet but we should if the patterns in this PR are something we want to catch on.